src/AdyenPaymentShopware6.php

```diff
@@ -85,57 +85,40 @@ public function uninstall(UninstallContext $uninstallContext): void
             return;
         }
 
-        //Search for config keys that contain the bundle's name
-        /** @var EntityRepositoryInterface $systemConfigRepository */
-        $systemConfigRepository = $this->container->get('system_config.repository');
-        $criteria = (new Criteria())
-            ->addFilter(
-                new ContainsFilter('configurationKey', ConfigurationService::BUNDLE_NAME . '.config')
-            );
-        $idSearchResult = $systemConfigRepository->searchIds($criteria, Context::createDefaultContext());
-
-        //Formatting IDs array and deleting config keys
-        $ids = \array_map(static function ($id) {
-            return ['id' => $id];
-        }, $idSearchResult->getIds());
-        $systemConfigRepository->delete($ids, Context::createDefaultContext());
-
-        //Dropping database tables
-        $tables = [
-            NotificationEntityDefinition::ENTITY_NAME,
-            PaymentStateDataEntityDefinition::ENTITY_NAME,
-            PaymentResponseEntityDefinition::ENTITY_NAME
-        ];
-        $connection = $this->container->get(Connection::class);
-        foreach ($tables as $table) {
-            $connection->executeUpdate(\sprintf('DROP TABLE IF EXISTS `%s`', $table));
-        }
+        $this->removePluginData();
     }
 
     public function update(UpdateContext $updateContext): void
     {
-        if (\version_compare($updateContext->getCurrentPluginVersion(), '1.2.0', '<')) {
+        $currentVersion = $updateContext->getCurrentPluginVersion();
+
+        if (\version_compare($currentVersion, '1.2.0', '<')) {
             $this->updateTo120($updateContext);
         }
 
-        if (\version_compare($updateContext->getCurrentPluginVersion(), '1.4.0', '<')) {
+        if (\version_compare($currentVersion, '1.4.0', '<')) {
             $this->updateTo140($updateContext);
         }
+
+        if (\version_compare($currentVersion, '1.6.0', '<')) {
+            $this->updateTo160($updateContext);
+        }
     }
 
     private function addPaymentMethod(PaymentMethodInterface $paymentMethod, Context $context): void
     {
-        $paymentMethodExists = $this->getPaymentMethodId($paymentMethod->getPaymentHandler());
-
-        // Payment method exists already, no need to continue here
-        if ($paymentMethodExists) {
-            return;
-        }
+        $paymentMethodId = $this->getPaymentMethodId($paymentMethod->getPaymentHandler());
 
         /** @var PluginIdProvider $pluginIdProvider */
         $pluginIdProvider = $this->container->get(PluginIdProvider::class);
         $pluginId = $pluginIdProvider->getPluginIdByBaseClass(get_class($this), $context);
 
+        // Payment method exists already, set the pluginId
+        if ($paymentMethodId) {
+            $this->setPluginId($paymentMethodId, $pluginId, $context);
```

Comment:
pluginId needs to be set in the case of a merchant reinstalling the plugin, otherwise all redirect PMs will not work

Reply:
Nice!